packages/flutter/lib/src/material/bottom_navigation_bar.dart

```diff
@@ -907,6 +920,20 @@ class _BottomNavigationBarState extends State<BottomNavigationBar> with TickerPr
         widget.unselectedFontSize,
       );
 
+    final IconThemeData effectiveSelectedIconTheme =
+      _effectiveIconTheme(
+        widget.selectedIconTheme ?? bottomTheme.selectedIconTheme,
+        widget.selectedItemColor,
+        widget.selectedLabelStyle,
```

Comment:
Why are we using  the label style's color for the icon? It describes the text color, not the icon color.

Reply:
@darrenaustin  You are correct. I don't know how I got confused. Will make changes and look again at the color tween.

Reply:
@darrenaustin Fixed by removing the dependency on the text style, but I was thinking more about this. 
Shouldn't i make the effectiveIconTheme depend on bottomTheme and themeColor, so it ends up like this
```dart
    final IconThemeData effectiveSelectedIconTheme =
      _effectiveIconTheme(
        widget.selectedIconTheme ?? bottomTheme.selectedIconTheme,
        widget.selectedItemColor 
        ?? bottomTheme.selectedItemColor 
        ?? themeColor
      );
 ```
 
Unfortunately this also breaks the tests for the shifting backgroundColor